### PR TITLE
Vista

### PR DESCRIPTION
**Pull request by SunilSimha**
- Including the VISTA survey into the list of allowed optical surveys.
- Adding a bunch of catalog merging functions and a function to search all available surveys at once.
- Tweaking CIGALE and EAZY wrappers.
- Fixing a Pan-STARRS bug that resulted in duplicate entries in a PS1 search catalog.
- Updated survey unit tests. All run fine on my local machine.

### CODE REVIEW

File: frb/surveys/catalog_utils.py [hunk 1]
```diff
@@ -388,7 +389,85 @@ def convert_mags_to_flux(photometry_table, fluxunits='mJy'):
         fluxtable[mag][uplimit] = fluxtable[mag][uplimit] 
 
     return fluxtable
+
+def remove_duplicates(tab:Table, idcol:str)->Table:
+    """
+    In an astropy table if there are duplicate
```
Comment (profxj) [inline frb/surveys/catalog_utils.py:395]: Comment on what is duplicated.  Is it just ID?

File: frb/surveys/catalog_utils.py [hunk 2]
```diff
@@ -388,7 +389,85 @@ def convert_mags_to_flux(photometry_table, fluxunits='mJy'):
...
+    tab.sort(idcol)
+    # Get the duplicates.
+    duplicate_ids = np.where(tab[1:][idcol]==tab[:-1][idcol])[0]+1
+    tab.remove_rows(duplicate_ids)
+    return tab
+    
+def xmatch_and_merge_cats(tab1:Table, tab2:Table, tol:units.Quantity=1*units.arcsec,
+                        table_names:tuple=('1','2'), **kwargs)->Table:
+    """
+    Given two source catalogs, cross-match and merge them. This function
```
Comment (profxj) [inline frb/surveys/catalog_utils.py:418]: Maybe comment the Tables need to have `ra` and `dec` columns.

File: frb/surveys/catalog_utils.py [hunk 3]
```diff
@@ -388,7 +389,85 @@ def convert_mags_to_flux(photometry_table, fluxunits='mJy'):
...
+    """
+    if table_names is not None:
+        assert len(table_names)==2, "Invalid number of table names for two tables."
+        assert (type(table_names[0])==str)&(type(table_names[1])==str), "Table names should be strings."
     
+    assert np.all(np.isin(['ra','dec'],tab1.colnames)), "Table 1 doesn't have column 'ra' and/or 'dec'."
+    assert np.all(np.isin(['ra','dec'],tab2.colnames)), "Table 2 doesn't have column 'ra' and/or 'dec'."
+
+
+    matched_tab1, matched_tab2 = xmatch_catalogs(tab1, tab2, tol, **kwargs)
```
Comment (profxj) [inline frb/surveys/catalog_utils.py:440]: Add a few more inline comments here and there.

File: frb/surveys/survey_utils.py [hunk 4]
```diff
@@ -114,10 +122,85 @@ def in_which_survey(coord:SkyCoord)->dict:
...
+                # First time
+                combined_cat = survey.catalog
+            else:
+                # Combine otherwise
+                # TODO: Need to deal with duplicate column names more elegantly.
+                combined_cat = xmatch_and_merge_cats(combined_cat, survey.catalog,)
+        # No objects found?
+        elif len(survey.catalog)==0:
+            print("Empty table in "+surveyname)
+            # Proper empty table
```
Comment (profxj) [inline frb/surveys/survey_utils.py:194]: remove this?

**Review by profxj — APPROVED** [review]
Great stuff. And thanks for the tests.
Just a few requests from me.

File: frb/surveys/catalog_utils.py (hunk 1 above)
Comment (SunilSimha) [reply frb/surveys/catalog_utils.py:395]: Add a line clarifying this.

File: frb/surveys/catalog_utils.py (hunk 2 above)
Comment (SunilSimha) [reply frb/surveys/catalog_utils.py:418]: Done.

File: frb/surveys/catalog_utils.py (hunk 3 above)
Comment (SunilSimha) [reply frb/surveys/catalog_utils.py:440]: Done.

File: frb/surveys/survey_utils.py (hunk 4 above)
Comment (SunilSimha) [reply frb/surveys/survey_utils.py:194]: Done.

**Pull request merged**